### PR TITLE
Add tunnel config

### PR DESCRIPTION
**Pull request by sojindal**
Add config to create tunnel (vxlan or gre)

### CODE REVIEW

File: sonic_api.yaml [hunk 1]
```diff
@@ -1400,3 +1604,24 @@ definitions:
...
+      description:
+        type: string
+      src_ip:
+        type: string
+        format: ipv4
+      vrf_id:
+        type: string
+        description: id representing the virtual routing instance (VRF) this tunnel is associated to.
+      encap_src:
+        type: string
```
Comment (prsunny) [inline sonic_api.yaml:1621]: What is this field? Is it encap src ip?

File: sonic_api.yaml [hunk 2]
```diff
@@ -1400,3 +1604,24 @@ definitions:
...
+  TunnelEntry:
+    type: object
+    properties:
+      tunnel_name:
+        type: string
+      description:
+        type: string
+      src_ip:
+        type: string
+        format: ipv4
```
Comment (prsunny) [inline sonic_api.yaml:1616]: Why only ipv4?

File: sonic_api.yaml [hunk 3]
```diff
@@ -1255,6 +1255,210 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 #----------------------------------------------
+# Tunnel encap object for vxlan and gre tunnels
+#----------------------------------------------
+  '/config/tunnel/encap/{tunnel_type}/{tunnel_name}':
+    put:
+      summary: Create a tunnel
+      description: If a tunnel with this name doesn't exist yet then create a new tunnel with these parameters. If the tunnel already exists, it's attributes will be updated.
```
Comment (prsunny) [inline sonic_api.yaml:1263]: Please mention, which attributes can be modified.

File: sonic_api.yaml (hunk 2 above)
Comment (sojindal) [reply sonic_api.yaml:1616]: We will add ipv6 support in later iterations.

File: sonic_api.yaml (hunk 1 above)
Comment (sojindal) [reply sonic_api.yaml:1621]: Yes, it could be the encap source ip or the loopback interface.

File: sonic_api.yaml (hunk 3 above)
Comment (sojindal) [reply sonic_api.yaml:1263]: I don't see anything which cannot be updated in TunnelEntry object. Exceptions can be added in implementation for lets say if some field updation is dependent on some other resources.

**Review by prsunny — APPROVED** [review]

**Pull request merged**